lib/livebook_web/live/session_live.ex

```diff
@@ -1425,6 +1471,47 @@ defmodule LivebookWeb.SessionLive do
     )
   end
 
+  defp prune_outputs_in_data(data) do
+    %{
+      data
+      | notebook:
+          Notebook.update_cells(data.notebook, fn
+            %Cell.Elixir{} = cell ->
+              %{cell | outputs: prune_outputs(cell.outputs)}
+
+            cell ->
+              cell
+          end)
+    }
+  end
+
+  defp prune_outputs(outputs) do
+    outputs
+    |> Enum.reverse()
+    |> do_prune_outputs()
+    |> Enum.reverse()
```

Comment:
I would make this tail recursive so you can skip the last reverse:

```suggestion
    |> do_prune_outputs([])
```